### PR TITLE
chore: Create toolbar widget

### LINKED ISSUE #30439: Create toolbar button group widget
Design reference doc - https://www.notion.so/appsmith/Button-group-widget-design-proposal-d84617fa456f4adea6b69cd7e77e7936#3f481fd5b6fc479e892d64cd3963c791

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #30439 

## Summary by CodeRabbit

- **New Features**
    - Enhanced feature flag checks to directly return `true` for specific flag names, improving efficiency.
    - Updated the `ButtonListControl` component to support separators between buttons for better visual distinction.

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx [hunk 1]
```diff
@@ -12,14 +12,20 @@ export const MenuList = <T extends object>(props: MenuListProps<T>) => {
...
     <ul {...menuProps} className={listClassName} ref={ref}>
-      {[...state.collection].map((item) => (
-        <MenuItem
-          className={itemClassName}
-          item={item}
-          key={item.key}
-          state={state}
-        />
-      ))}
+      {[...state.collection].map((item) => {
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx:15]: Why do we need this for Menu?

File: app/client/src/ce/selectors/featureFlagsSelectors.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ export const selectFeatureFlagCheck = (
   flagName: FeatureFlag,
 ): boolean => {
   const flagValues = selectFeatureFlags(state);
+
+  if (flagName === "ab_wds_enabled") return true;
```
Comment (KelvinOm) [inline app/client/src/ce/selectors/featureFlagsSelectors.ts:15]: Don't forget to delete this.

File: app/client/src/components/propertyControls/ButtonListControl.tsx [hunk 3]
```diff
@@ -95,15 +99,26 @@ class ButtonListControl extends BaseControl<ControlProps, State> {
...
-        <Button
-          className="self-end"
-          kind="tertiary"
-          onClick={this.addOption}
-          size="sm"
-          startIcon="plus"
-        >
-          Add new button
-        </Button>
+        <Flex justifyContent="end">
```
Comment (KelvinOm) [inline app/client/src/components/propertyControls/ButtonListControl.tsx:102]: Does this affect current apps(fixed and auto layout)?

Comment (jsartisan) [reply app/client/src/components/propertyControls/ButtonListControl.tsx:102]: No. It wont affect.

File: app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx (hunk 1 above)
Comment (jsartisan) [reply app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx:15]: I added it so that we can group menu items too.

Comment (jsartisan) [reply app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx:15]: Right now you will see the separator item in menu dropdown but it won't see anything as it does not have any styles.

File: app/client/src/components/propertyControls/ButtonListControl.tsx [hunk 4]
```diff
@@ -95,15 +99,26 @@ class ButtonListControl extends BaseControl<ControlProps, State> {
...
+          {this.props.allowSeparators && (
+            <Button
+              className="self-end"
+              kind="tertiary"
+              onClick={() => this.addOption({ isSeparator: true })}
+              size="sm"
+            >
+              Add Separator
```
Comment (KelvinOm) [inline app/client/src/components/propertyControls/ButtonListControl.tsx:110]:
```suggestion
              Add separator
```

File: app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/headless/src/components/Menu/src/MenuList.tsx:15]: Since we don't need this yet, it's better to delete it.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**